src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -137,3 +147,97 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
         if comment:
             query += f" comment='{comment}'"
         return self._execute_query(query)
+
+    def execute(
+        self,
+        stage_path: str,
+        on_error: OnErrorType,
+        parameters: Optional[List[str]] = None,
+    ):
+        files_list = self._get_files_list_from_stage(stage_path)
+        filtered_list = self._filter_files_list(stage_path, files_list)
+        sorted_list = sorted(
+            filtered_list, key=cmp_to_key(self._stage_files_comparator)
+        )
+
+        sql_parameters = self._parse_execute_parameters(parameters)
+        results = []
+        for file in sorted_list:
+            results.append(self._call_execute_immediate(file, sql_parameters, on_error))
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = stage_path.split("/")[0]
+        files_list_result = self.list_files(stage_name).fetchall()
+
+        if len(files_list_result) == 0:
+            raise ClickException(f"No files found on stage '{stage_name}'")
+
+        return [f["name"] for f in files_list_result]
+
+    def _filter_files_list(
+        self, stage_path: str, files_on_stage: List[str]
+    ) -> List[str]:
+        if stage_path.startswith("@"):
+            stage_path = stage_path[1:]
+
+        if not stage_path.__contains__("/"):
+            filtered_list = files_on_stage
+        elif has_magic(stage_path):
```

Comment:
```suggestion
        elif glob.has_magic(stage_path):
```
Consider doing this way to make it easier to understand

Reply:
Agree